core-bundle/src/Routing/RouteProvider.php

```diff
@@ -338,7 +338,9 @@ private function getPageIdsFromNames(array $names): array
         $ids = [];
 
         foreach ($names as $name) {
-            if (0 !== strncmp($name, 'tl_page.', 8)) {
+            $parts = explode('.', $name);
+
+            if ('tl_page' !== ($parts[0] ?? '') || 'error_404' === ($parts[2] ?? '')) {
```

Comment:
```suggestion
            if ('tl_page' !== $parts[0] || 'error_404' === ($parts[2] ?? '')) {
```

`$parts[0]` will always be set, even if the string is empty or does not contain the split character (see https://3v4l.org/O2Qvv).